WIP: bluealsa-aplay man page update (#572)
Arkq/bluez-alsa: doc/bluealsa-aplay.1.rst

example:

::

    $ aplay -l
    **** List of PLAYBACK Hardware Devices ****
    card 0: PCH [HDA Intel PCH], device 0: ALC236 Analog [ALC236 Analog]
      Subdevices: 1/1
      Subdevice #0: subdevice #0
    card 1: Device [USB Audio Device], device 0: USB Audio [USB Audio]
      Subdevices: 1/1
      Subdevice #0: subdevice #0

Here the first word after the card number is the card name, so that to use the
HDA Intel PCH card:

::

    bluealsa-aplay -D default:PCH

or to use the USB card:

::

    bluealsa-aplay -D default:Device

Some sound cards offer more than one playback PCM device. ALSA identifies these
devices by using an index number for each. Such a card might produce **aplay -l**
output like:

::

    $ aplay -l
    **** List of PLAYBACK Hardware Devices ****
    card 0: PCH [HDA Intel PCH], device 0: ALC236 Analog [ALC236 Analog]
      Subdevices: 1/1
      Subdevice #0: subdevice #0
    card 0: PCH [HDA Intel PCH], device 3: HDMI 0 [HDMI 0]
      Subdevices: 1/1
      Subdevice #0: subdevice #0
    card 0: PCH [HDA Intel PCH], device 7: HDMI 1 [HDMI 1]
      Subdevices: 1/1
      Subdevice #0: subdevice #0

> Here the device index number is given by ``device N``, so in this example
> the Analog output could be used with:
> 
> ::
> 
>     bluealsa-aplay -D plughw:PCH,0
> 
> and the first HDMI port (``HDMI 0``) is used with:
> 
> ::
> 
>     bluealsa-aplay -D plughw:PCH,3
> 

Author: ```suggestion
```

For most distributions, the installed definition of ``default`` for most sound
cards will automatically convert the samplerate and format of the BT audio to those
supported by the playback card, using ALSA's  ``plughw`` plug-in.  ``default`` also refers
to device 0 so we have to explicitly use the ``plughw`` plugin for other devices to get
automatic samplerate and format conversion.

The ``default`` for each sound card often also includes ALSA's ``dmix`` plug-in so that
multiple sound sources (from **bluealsa-aplay** and/or other applications) can play
simultaneously to the same card. It needs an additional configuration file to use both
``dmix`` and ``plughw`` with a device other than 0. The file should be similar to the example
for **DMIX** below, substituting the card number and device number for ``hw:0,0``.

> **bluealsa-aplay** does not perform any mixing of streams. If multiple bluetooth
> devices are connected it opens a new connection to the ALSA PCM device for each
> stream. Therefore the ALSA playback PCM must itself allow multiple open
> connections and mix the streams together (see option **--single-audio** to
> change this behavior). For this the ALSA ``dmix`` plugin must be used.
> 
> If ``dmix`` is not provided by ``default`` in your distribution or if you have a special
> case, such as selecting a specific subset of channels from a multi-channel device or
> to duplicate the stream across multiple output devices etc. you will need some
> additional configuration. It is recommended to seek advice from your distribution
> as this can be quite complex and is beyond the scope of this manual.

Author: ```suggestion
There will be some cases where the ``default`` device is not sufficient. These
are generally special cases, such as selecting a specific subset of channels
from a multi-channel device or to duplicate the stream across multiple output
devices etc. These cases generally require some additional configuration by the
user, and it is recommended to seek advice from your distribution as this can be
quite complex and is beyond the scope of this manual.
```

        }
    }

EXAMPLES
========

> 

Author: > I would use --mixer-device=hw:USB to be consistent with the pcm naming. Also, it might be useful to preface this with a note that these names are just examples.

Yes, that will be better indeed. But (and here is the limit of my ALSA knowledge) how to list available mixer names? In `alsamixer` the case is simple, there is a combobox with available hardware mixers ....but `amixer` can not handle these names as a valid mixer device. Or simply mixer device is in fact a card name?

> A list of control names for a card called "USB" can be obtained by running `amixer -c USB scontrols`

OK, such info could be useful.

> 

Reviewer: > amixer can not handle these names as a valid mixer device

So, amixer (and alsamixer) can be given a mixer either by using the card name as in `-c NAME`, which is clearly only valid for sound cards, or by using the device name, as in `-D DEVICE_NAME`. A device name is a configuration entry such as `hw:CARD=USB`, `bluealsa:DEV=00:00:00:00:00:00`, etc, with the usual argument expansion. Clearly device name is more general, but is also *much* more difficult to explain to ALSA novices. There is no utility from the ALSA project that lists CTL devices on the command line. I have my own here: https://github.com/borine/alsa-utils-completions because I was unable to find any ready-made example.

> 

Author: I've added a little bit rephrased your suggestion as a preface to command line examples. So, right now in the example we've got mentioned `aplay -l` which might be a good "tip" for a newbie to ALSA as well.

> 

Reviewer: That looks good to me. FWIW card names can also be obtained by
```
cat /proc/asound/cards
```
or
```
cat /sys/class/sound/card*/id
```
of which the second gives a simple list of names with no need to parse the output, but also no clue as to which name refers to which card .
